src/systems/personal-space-bubble.js

```diff
@@ -73,36 +73,47 @@ AFRAME.registerSystem("personal-space-bubble", {
     }
   },
 
-  tick() {
-    if (!this.data.enabled) return;
+  tick: (function() {
+    const tempInvasionFlags = [];
 
-    // precondition for this stuff -- the bubbles and invaders need updated world matrices.
-    // right now this is satisfied because we update the world matrices in the character controller
+    return function() {
+      if (!this.data.enabled) return;
 
-    for (let i = 0; i < this.invaders.length; i++) {
-      this.invaders[i].setInvading(false);
-    }
+      tempInvasionFlags.length = 0;
 
-    // Loop through all of the space bubbles (usually one)
-    for (let i = 0; i < this.bubbles.length; i++) {
-      const bubble = this.bubbles[i];
+      // precondition for this stuff -- the bubbles and invaders need updated world matrices.
+      // right now this is satisfied because we update the world matrices in the character controller
+      for (let i = 0; i < this.invaders.length; i++) {
+        tempInvasionFlags[i] = false;
+      }
+
+      // Loop through all of the space bubbles (usually one)
+      for (let i = 0; i < this.bubbles.length; i++) {
+        const bubble = this.bubbles[i];
 
-      bubblePos.setFromMatrixPosition(bubble.el.object3D.matrixWorld);
+        bubblePos.setFromMatrixPosition(bubble.el.object3D.matrixWorld);
```

Comment:
Need to call `bubble.el.object3D.updateMatrices()` and `invader.el.object3D.updateMatrices()` before accessing `matrixWorld`s